namenode/src/main/java/com/ibm/crail/namenode/BlockStore.java

```
@@ -46,13 +46,23 @@ public BlockStore(){
 		}		
 	}
 
-	public short addBlock(BlockInfo blockInfo) throws UnknownHostException {
+	public short addBlock(NameNodeBlockInfo blockInfo) throws UnknownHostException {
 		int storageClass = blockInfo.getDnInfo().getStorageClass();
 		return storageClasses[storageClass].addBlock(blockInfo);
 	}
 
-	public BlockInfo getBlock(int storageClass, int locationAffinity) throws InterruptedException {
-		BlockInfo block = null;
+	public boolean regionExists(BlockInfo region) {
+		int storageClass = region.getDnInfo().getStorageClass();
+		return storageClasses[storageClass].regionExists(region);
+	}
+
+	public short updateRegion(BlockInfo region) {
+		int storageClass = region.getDnInfo().getStorageClass();
+		return storageClasses[storageClass].updateRegion(region);
+	}
+
+	public NameNodeBlockInfo getBlock(int storageClass, int locationAffinity) throws InterruptedException {
+		NameNodeBlockInfo block = null;
 		if (storageClass > 0){
 			if (storageClass < storageClasses.length){
 				block = storageClasses[storageClass].getBlock(locationAffinity);
```

Do we want to throw an exception if the storage class is invalid?